server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -102,6 +114,13 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
   {
     Collection<String> dataSourcesToKill =
         params.getCoordinatorDynamicConfig().getSpecificDataSourcesToKillUnusedSegmentsIn();
+    double killTaskSlotRatio = params.getCoordinatorDynamicConfig().getKillTaskSlotRatio();
+    int maxKillTaskSlots = params.getCoordinatorDynamicConfig().getMaxKillTaskSlots();
+    int availableKillTaskSlots = getAvailableKillTaskSlots(killTaskSlotRatio, maxKillTaskSlots);
+    if (0 == availableKillTaskSlots) {
+      log.warn("Not killing any unused segments because there are no available kill task slots at this time.");
```

Comment:
Does this need to be a warning? It seems like it could be quite verbose if there are long running kill tasks and the KillUnusedSegments duty is scheduled to run frequently. 

Perhaps, a better way of surfacing this information is emitting the number of kill tasks and calculated kill capacity as metrics so that operators can look at these metrics to decide if they need more kill capacity.

Reply:
changed to debug. 💯  about adding metrics, will do that in a follow up patch.